### ISSUE TITLE
[DependencyInjection] ParameterCircularReferenceException when creating secrets

### ISSUE THREAD

**Issue by jderusse** [symfony/symfony#37281]
**Symfony version(s) affected**: 5.1

**Description**  
When creating secrets with command, a `ParameterCircularReferenceException` is triggered.

**How to reproduce**  
```cli
symfony new project --full
bin/console secret:set --env=prod TEST
bin/console secret:set --env=prod TEST2
```

**Additional context**  

The EnvVarProcessor contains code to rpevend it, but I don't understand why it fails. Maybe related to the `LazyString` and exception triggered in `__toString`.
stack trace show that exception is triggered throught EnvVarProcessor line 153.
Line this line is wrap in try/catch block
```
#0 /data/oss/project/vendor/symfony/string/LazyString.php(41): 
...
#6 /data/oss/project/vendor/symfony/dependency-injection/EnvVarProcessor.php(153): 
```
https://github.com/symfony/symfony/blob/b04512a6fab1f69c3a7dec09e82000cf150d9672/src/Symfony/Component/DependencyInjection/EnvVarProcessor.php#L145-L164

/cc @nicolas-grekas

**Comment by simonberger** [comment]
I failed to reproduce this. Exception is thrown and caught.
I'm currently on a windows machine with PHP 7.4 in case this matters.

**Comment by jderusse** [comment]
currently using php 7.3

**Comment by picks44** [comment]
I have the issue and using php 7.2

**Comment by weaverryan** [comment]
I personally failed to reproduce this, but a user on SymfonyCasts definitely hit it - https://symfonycasts.com/screencast/symfony5-upgrade/secrets-vault-setup#comment-4957289406

So I'd say that there IS a legit issue here. The circular reference in that case was for `SYMFONY_DECRYPTION_SECRET` which the user never defined anywhere. I assume @jderusse that the `ParameterCircularReferenceException` was also for `SYMFONY_DECRYPTION_SECRET` in your case?

**Comment by simonberger** [comment]
This issue should be closed as a duplicate of #37142. Discussion and Information are splitting.

**Comment by jderusse** [comment]
@weaverryan php 7.4 I guess

closing in favor of #37142
